src/thermo/ConstDensityTabulatedThermo.cpp

```diff
@@ -0,0 +1,203 @@
+/**
+ *  @file ConstDensityTabulatedThermo.cpp
+ * Definition file for a derived class of ThermoPhase that assumes either
+ * an ideal gas or ideal solution approximation and handles
+ * variable pressure standard state methods for calculating
+ * thermodynamic properties (see \ref thermoprops and
+ * class \link Cantera::ConstDensityTabulatedThermo ConstDensityTabulatedThermo\endlink).
+ */
+/*
+ * Copyright (2005) Sandia Corporation. Under the terms of
+ * Contract DE-AC04-94AL85000 with Sandia Corporation, the
+ * U.S. Government retains certain rights in this software.
+ */
+
+#include "cantera/thermo/ConstDensityTabulatedThermo.h"
+#include "cantera/base/ctml.h"
+
+namespace Cantera
+{
+
+void ConstDensityTabulatedThermo::setMoleFractions(const doublereal* const x)
+{
+	Phase::setMoleFractions(x);
+	_updateThermo();
+}
+
+void ConstDensityTabulatedThermo::setMoleFractions_NoNorm(const doublereal* const x)
+{
+	Phase::setMoleFractions_NoNorm(x);
+	_updateThermo();
+}
+
+void ConstDensityTabulatedThermo::setMassFractions(const doublereal* const y)
+{
+	Phase::setMassFractions(y);
+	_updateThermo();
+}
+
+void ConstDensityTabulatedThermo::setMassFractions_NoNorm(const doublereal* const y)
+{
+	Phase::setMassFractions_NoNorm(y);
+	_updateThermo();
+}
+
+void ConstDensityTabulatedThermo::setConcentrations(const doublereal* const conc)
+{
+	Phase::setConcentrations(conc);
+	_updateThermo();
+}
+
+void ConstDensityTabulatedThermo::_updateThermo()
+{
+	doublereal tnow = temperature();
+	doublereal xnow = moleFraction(m_kk_mod);
+	doublereal c[4];
+	doublereal dS_corr = 0.0;
+	doublereal tlow = 0.0, thigh = 0.0;
+	int type = 0;
+	if (m_tlast != tnow || m_xlast != xnow) {
+		c[0]=tnow;

```

Comment:
Please be consistent with spaces around `=` in the code here. I think the preference in C++ code is to include the spaces.